execute/metrics/prom.go

```diff
@@ -0,0 +1,189 @@
+package metrics
```

Comment:
This is not changed, it was moved to a separate file to reflect the commit's structure